geniusyield-market-maker/src/GeniusYield/MarketMaker/Utils.hs

```diff
@@ -53,3 +66,132 @@ instance HasPartialOrderNftScript DEXInfo where
 
 instance HasPartialOrderConfigAddr DEXInfo where
   getPartialOrderConfigAddr = dexPartialOrderConfigAddr
+
+
+-------------------------------------------------------------------------------
+-- Common Price Resolution
+-------------------------------------------------------------------------------
+
+-- | Generic time resolution for OHLC Candles
+data CommonResolution = CRes5m | CRes15m | CRes30m | CRes1h | CRes4h
+                      | CRes1d | CRes1w  | CRes1mo
+                      deriving stock (Eq, Ord, Show, Generic)
+
+instance FromJSON CommonResolution
+
+class PriceResolution a where
+  resolutionInherited :: Map.Map CommonResolution a
+
+instance PriceResolution Maestro.Resolution where
+  resolutionInherited = Map.fromList [ (CRes5m, Res5m), (CRes15m, Res15m), (CRes30m, Res30m), (CRes1h, Res1h)
+                                     , (CRes4h, Res4h), (CRes1d, Res1d), (CRes1w, Res1w), (CRes1mo, Res1mo) ]
+
+fromCommonResolution :: PriceResolution a => CommonResolution -> IO a
+fromCommonResolution = maybe (throwIO $ userError "Undefined common price resolution.") pure
+                       . flip Map.lookup resolutionInherited
+
+
+-------------------------------------------------------------------------------
+-- TapTools OHLCV query
+-------------------------------------------------------------------------------
+
+-- | Taptools time resolution for OHLC Candles
+data TtResolution = TtRes3m | TtRes5m | TtRes15m | TtRes30m | TtRes1h | TtRes2h | TtRes4h | TtRes12h
+                  | TtRes1d | TtRes3d | TtRes1w  | TtRes1mo
+                  deriving stock (Eq, Ord, Generic)
+
+instance FromJSON TtResolution
+
+instance Show TtResolution where
+  show = let kvm = Map.fromList [ (TtRes3m, "3m"), (TtRes5m, "5m"), (TtRes15m, "15m"), (TtRes30m, "30m")
+                                , (TtRes1h, "1h"), (TtRes2h, "2h"), (TtRes4h, "4h"), (TtRes12h, "12h")
+                                , (TtRes1d, "1d"), (TtRes3d, "3d"), (TtRes1w, "1w"), (TtRes1mo, "1M") ]
+         in  fromJust . flip Map.lookup kvm
```

Comment:
Use of `fromJust` appears unnecessary if done via pattern match. Have a look on how I wrote show instance for resolution [here](https://github.com/maestro-org/haskell-sdk/blob/609515b3f8861efaa17adf4e599aa1b845775a6d/src/Maestro/Types/V1/DefiMarkets.hs#L50-L53).

Reply:
However I don't think `fromJust` is problematic here, meaning that it does not lead to a partial function.  Because every possible value of `TtResolution` is a key in the given key-value-map `kvm`.

Reply:
If a new data constructor is added, it would create issue.

Reply:
You're right.  This now has been fixed using pattern match.  I could not use the neat trick [`toConstr >>> show >>> drop 5`](https://github.com/maestro-org/haskell-sdk/blob/609515b3f8861efaa17adf4e599aa1b845775a6d/src/Maestro/Types/V1/DefiMarkets.hs#L50-L53) because pattern breaks in the month case (unless I change the constructor name, but I don't want to do that for consistency with Maestro constructor's name).

Reply:
Done using pattern match.